libraries/engage/components/ScrollHeader/index.jsx

```diff
@@ -1,56 +1,51 @@
 import React, {
-  useState, useContext, useRef, useCallback, useEffect,
+  useState, useCallback, useEffect,
 } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import get from 'lodash/get';
-import { useScroll, hasWebBridge } from '@shopgate/engage/core';
-import { header, hidden } from './style';
-import { ViewContext } from '../View';
+import { viewScroll$ } from '@shopgate/engage/core/streams';
+import {
+  root, scrolledIn, scrolledOut, transition,
+} from './style';
 
 /**
  * Scroll Header component
  * @param {Object} props props
  * @returns {JSX}
  */
-function ScrollHeader({ className, children }) {
-  const ref = useRef();
+function ScrollHeader({
+  className, children, hideOnScroll, scrollOffset,
+}) {
   const [shouldHideHeader, setShouldHideHeader] = useState(false);
-  const { contentRef } = useContext(ViewContext);
-  const [offset, setOffset] = useState(0);
-  const [offsetTop, setOffsetTop] = useState(0);
 
-  const onScroll = useCallback((callbackData) => {
-    const { previousScrollTop, currentScrollTop } = callbackData;
-    if (previousScrollTop !== currentScrollTop) {
-      const isScrolledDown = previousScrollTop < currentScrollTop;
-      const box = ref.current.getBoundingClientRect();
-      const stickHeader = currentScrollTop >= offset + 100;
-      if (!previousScrollTop) {
-        setOffset(currentScrollTop + box.top);
+  const onScroll = useCallback((scrollEvent) => {
+    const {
+      scrollTop, scrolled, scrollOut, scrollIn,
+    } = scrollEvent;
+
+    if (scrolled) {
```

Comment:
instead of these nested if cases, would it be better to start with checking 
if (!scrolled) return;
?

Reply:
done